src/python_framework_v02/troute/network/reservoirs/levelpool/levelpool.pyx

```diff
@@ -2,32 +2,44 @@ cimport numpy as np
 import numpy as np
 from libc.stdlib cimport malloc, free
 
+from troute.network.reach cimport compute_type, _Reach
 """
 Externally defined symbols
 """
-############ Level Pool Reservoir Interface ############
-cdef extern void* get_lp_handle() nogil;
 
-cdef extern void init_lp(void* handle, float *water_elevation, float *lake_area, float *weir_elevation,
-                    float *weir_coefficient, float *weir_length, float *dam_length, float *orifice_elevation,
-                    float *orifice_coefficient, float *orifice_area, float *max_depth, int *lake_number) nogil;
+cdef extern from "levelpool_structs.c":
+  void init_levelpool_reach(_Reach* reach, int lake_number,
+                            float dam_length, float area, float max_depth,
+                            float orifice_area, float orifice_coefficient, float orifice_elevation,
+                            float weir_coefficient, float weir_elevation, float weir_length,
+                            float initial_fractional_depth, float water_elevation
+  )
+  void free_levelpool_reach(_Reach* reach)
+  void route(_Reach* reach, float routing_period, float inflow, float lateral_inflow, float* outflow,  float* water_elevation) nogil
 
-cdef extern void run_lp(void* handle, float *inflow, float *lateral_inflow,
-                    float *water_elevation, float *outflow, float *routing_period) nogil;
+cdef void run(_Reach* reach, float inflow, float lateral_inflow, float routing_period, float* outflow,  float* water_elevation) nogil:
+    route(reach, inflow, lateral_inflow, routing_period, outflow, water_elevation)
 
-cdef extern void free_lp(void* handle);
-
-cdef class MC_Levelpool(MC_Reach_Base_Class):
+cdef class MC_Levelpool(Reach):
   """
     MC_Reservoir is a subclass of MC_Reach_Base_Class
   """
 
-  def __init__(self, lake_number, long[::1] upstream_ids, args):
+  def __init__(self, long id, int lake_number, long[::1] upstream_ids, args):
     """
       Construct the kernel based on passed parameters,
       which only constructs the parent class
     """
-    super().__init__(upstream_ids)
+    super().__init__(id, upstream_ids, compute_type.RESERVOIR_LP)
+    #init the backing struct, pass a dam_length of 10.0 for now
+    #pass a negative water elevation, which causes the init to use the wrf hydro equation
+    #TODO put in __calloc__
```

Comment:
Some issues with `__calloc__`:  The python type isn't guaranteed to be properly constructed, so cannot depend on super class being constructured.  Thus I don't think we can put these C init functions in `__calloc__`, at least not in all cases.

Reply:
added as comment to code explaining why calloc wan't used
0440cfb750fc05cf2bdca9b0bbd66183302c3671